# ***Exploratory Data Analisys - Análisis sobre los hábitos de vida del ciudadano***
## ➡️ **Mapeo y descodificación del Dataset original**
------

### 🗒️**Bloque de contenidos:**

1. [Introducción](#Introducción)
2. [Carga de datos](#Carga-de-datos)  
    2.1. [Importación de módulos](#Importación-de-módulos)  
    2.2. [Importación de datos](#Importación-de-datos)  
    2.3. [Mapeo de las variables](#Mapeo-de-las-variables)  


### **1. Introducción**

El presente notebook realiza un *Análisis Exploratorio de Datos (EDA)* sobre un conjunto de datos de salud pública procedente del **Behavioral Risk Factor Surveillance System (BRFSS)**, una encuesta anual realizada por los **Centers for Disease Control and Prevention (CDC)** de Estados Unidos. Este dataset recoge información detallada relacionada con hábitos de vida, salud física, bienestar emocional, estilos de alimentación, sueño, ejercicio físico, consumo de alcohol y presencia de determinadas enfermedades.

El objetivo principal de este notebook es llevar a cabo un flujo completo y estructurado de **exploración, limpieza y visualización** básica de los datos, siguiendo el propósito del ejercicio práctico del máster. El análisis inicial permitirá comprender la estructura del dataset, evaluar su calidad, detectar problemas comunes (valores nulos, incoherencias, outliers, codificaciones irregulares) y preparar el conjunto de datos para análisis posteriores.

Este trabajo se centra exclusivamente en el EDA, sin aplicar técnicas de modelado ni análisis estadístico avanzado. La intención es **desarrollar criterio y orden** en las fases fundamentales de inspección y tratamiento de datos, aplicando buenas prácticas de análisis exploratorio.

A lo largo del notebook se realizarán los **siguientes pasos**:

- Carga del dataset y revisión general de su estructura.

- Exploración inicial: tipos de datos, valores nulos, duplicados, rangos y coherencia.

- Limpieza y normalización, justificando cada decisión.

- Visualizaciones básicas para comprender la distribución de variables clave.

- Conclusiones exploratorias que resuman los principales hallazgos.

El **objetivo final** es obtener una visión clara y ordenada del dataset, transformándolo desde su estado original hacia una versión depurada que facilite futuras tareas analíticas.

Para poder iniciar este EDA, se debe plantear una serie de preguntas que alce la iniciativa y la curiosidad del análisis, planteando estas **primeras cuestiones**:

- ¿Cómo se distribuyen las horas de sueño?

- ¿Qué prevalencia hay de mala salud mental o física?

- ¿Qué relación podría haber entre ejercicio, sueño y salud percibida?

- ¿Hay diferencias según sexo o grupo de edad?

---

### **2. Carga de datos**

#### ***2.1 Importación de módulos***



In [1]:
%pwd

'c:\\Users\\sorim\\Documents\\Data_Science\\Evolve\\repos\\projects-evolve\\EDA\\notebooks'

In [2]:
# Ajusto la ruta
%cd ..

c:\Users\sorim\Documents\Data_Science\Evolve\repos\projects-evolve\EDA


In [3]:
import scripts.variables
from scripts.funciones import mapeo_df_original
from scripts.modulos import *

#### ***2.2 Importación de datos***


In [4]:
df = pd.read_sas('./data/LLCP2022.XPT', format='xport', encoding='latin1')
df

_STATE  FMONTH     IDATE IMONTH IDAY IYEAR  DISPCODE       SEQNO  \
0          1.0     1.0  02032022     02   03  2022    1100.0  2022000001   
1          1.0     1.0  02042022     02   04  2022    1100.0  2022000002   
2          1.0     1.0  02022022     02   02  2022    1100.0  2022000003   
3          1.0     1.0  02032022     02   03  2022    1100.0  2022000004   
4          1.0     1.0  02022022     02   02  2022    1100.0  2022000005   
...        ...     ...       ...    ...  ...   ...       ...         ...   
445127    78.0    11.0  12192022     12   19  2022    1100.0  2022001527   
445128    78.0    11.0  12212022     12   21  2022    1100.0  2022001528   
445129    78.0    11.0  11292022     11   29  2022    1100.0  2022001529   
445130    78.0    11.0  12082022     12   08  2022    1100.0  2022001530   
445131    78.0    11.0  12142022     12   14  2022    1100.0  2022001531   

                _PSU  CTELENM1  ...  _SMOKGRP  _LCSREC  DRNKANY6  \
0       2.022000e+09       1.0  ...       4.0      NaN       2.0   
1       2.022000e+09       1.0  ...       4.0      NaN       2.0   
2       2.022000e+09       1.0  ...       4.0      NaN       2.0   
3       2.022000e+09       1.0  ...       3.0      2.0       2.0   
4       2.022000e+09       1.0  ...       4.0      NaN       1.0   
...              ...       ...  ...       ...      ...       ...   
445127  2.022002e+09       NaN  ...       4.0      NaN       7.0   
445128  2.022002e+09       NaN  ...       4.0      NaN       2.0   
445129  2.022002e+09       NaN  ...       1.0      NaN       7.0   
445130  2.022002e+09       NaN  ...       4.0      NaN       2.0   
445131  2.022002e+09       NaN  ...       3.0      NaN       1.0   

            DROCDY4_  _RFBING6      _DRNKWK2  _RFDRHV8  _FLSHOT7  _PNEUMO3  \
0       5.397605e-79       1.0  5.397605e-79       1.0       1.0       2.0   
1       5.397605e-79       1.0  5.397605e-79       1.0       2.0       2.0   
2       5.397605e-79       1.0  5.397605e-79       1.0       NaN       NaN   
3       5.397605e-79       1.0  5.397605e-79       1.0       9.0       9.0   
4       1.000000e+01       1.0  1.400000e+02       1.0       NaN       NaN   
...              ...       ...           ...       ...       ...       ...   
445127  9.000000e+02       9.0  9.990000e+04       9.0       NaN       NaN   
445128  5.397605e-79       1.0  5.397605e-79       1.0       NaN       NaN   
445129  9.000000e+02       9.0  9.990000e+04       9.0       2.0       2.0   
445130  5.397605e-79       1.0  5.397605e-79       1.0       1.0       1.0   
445131  5.000000e+01       2.0  9.990000e+04       9.0       NaN       NaN   

        _AIDTST4  
0            2.0  
1            2.0  
2            2.0  
3            2.0  
4            2.0  
...          ...  
445127       1.0  
445128       1.0  
445129       2.0  
445130       1.0  
445131       2.0  

[445132 rows x 328 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 445132 entries, 0 to 445131
Columns: 328 entries, _STATE to _AIDTST4
dtypes: float64(323), object(5)
memory usage: 1.1+ GB


Como se puede observar, es un dataset muy complejo en el que cuenta con 445.132 filas y 328 columnas. 

Al tener un tiempo limitado para poder analizar todos los datos, voy a centrarme en aquellas columnas que me pueden interesar más. 

Gracias a que la fuente de los datos también incluye un archivo donde viene información de cada una de las variables (incluido en la caperta data), me quedaré con:
- **Hábitos y estilo de vida**

    - `EXERANY2` -> ¿Realiza ejercicio?

    - `SLEPTIM1` -> Horas de sueño

    - `SMOKE100` -> Ha fumado 100 cigarrillos en su vida

    - `ALCDAY4` -> Frecuencia de consumo de alcohol

- **Estrés y salud mental**

    - `MENTHLTH` -> Días de mala salud mental (últimos 30 días)

- **Salud física**

    - `GENHLTH` -> Salud general

    - `PHYSHLTH` -> Días de mala salud física

    - `DIABETE4` -> Diabetes

    - `CVDCRHD4` -> Enfermedad cardiovascular

    - `ASTHMA3` -> Asma

    - `CHCOCNC1` -> Cáncer

- **Variables socio-demográficas**

    - `COLGSEX1` -> Género

    - `_AGEG5YR` -> Grupo de edad

    - `INCOME3` -> Nivel de ingresos

    - `_EDUCAG` -> Nivel educativo

    - `EMPLOY1` -> Empleo

    - `_STATE` -> Estado

    - `_BMI5CAT` -> Categprías de IMC

#### ***2.3 Mapeo de las variables***

Ahora que ya se ha hecho la selección de las variables que más me llaman la atención y las cuales quiero analizar, procedemos a mapear las variables ya que están codificada de manera numérica.

Para que sea más automatizado el mapeo de las variables, así como sus valores (ya que vienen codificadas de manera numérica), he creado un script de python para la ayuda de la conversión.


In [5]:
mapeo_df_original(df, scripts.variables)

EXERANY2 ASTHMA3 CHCOCNC1 COLGSEX1         _AGEG5YR  \
0            No      No       No      NaN  Age 80 or older   
1            No      No      Yes      NaN  Age 80 or older   
2           Yes      No       No      NaN     Age 55 to 59   
3           Yes     Yes       No      NaN              NaN   
4           Yes      No       No      NaN     Age 30 to 44   
...         ...     ...      ...      ...              ...   
445127      Yes     Yes       No      NaN     Age 18 to 24   
445128      Yes      No       No      NaN     Age 50 to 54   
445129       No      No       No      NaN     Age 65 to 69   
445130       No     Yes       No      NaN     Age 70 to 74   
445131      Yes      No       No      NaN     Age 30 to 44   

                       INCOME3                                     _EDUCAG  \
0                      Refused  Graduated from College or Technical School   
1         $25,000 to < $35,000                       Graduated High School   
2       $150,000 to < $200,000  Graduated from College or Technical School   
3          Don’t know/Not sure                       Graduated High School   
4         $25,000 to < $35,000        Attended College or Technical School   
...                        ...                                         ...   
445127       Less than $10,000                       Graduated High School   
445128    $50,000 to < $75,000  Graduated from College or Technical School   
445129     Don’t know/Not sure                       Graduated High School   
445130   $75,000 to < $100,000        Attended College or Technical School   
445131    $20,000 to < $25,000                Did not graduate High School   

                   EMPLOY1          _STATE       _BMI5CAT SLEPTIM1 SMOKE100  \
0                  Retired         Alabama            NaN      8.0       No   
1            Self-employed         Alabama     Overweight      6.0       No   
2                  Retired         Alabama     Overweight      5.0       No   
3                  Retired         Alabama  Normal Weight      7.0      Yes   
4              A homemaker         Alabama  Normal Weight      9.0       No   
...                    ...             ...            ...      ...      ...   
445127  Employed for wages  Virgin Islands     Overweight      6.0       No   
445128  Employed for wages  Virgin Islands     Overweight      7.0       No   
445129      Unable to work  Virgin Islands    Underweight      5.0      Yes   
445130             Retired  Virgin Islands          Obese      5.0       No   
445131  Employed for wages  Virgin Islands  Normal Weight      5.0      Yes   

                          ALCDAY4 MENTHLTH    GENHLTH PHYSHLTH DIABETE4  \
0       No drinks in past 30 days     None  Very good     None      Yes   
1       No drinks in past 30 days     None  Excellent     None       No   
2       No drinks in past 30 days      3.0  Very good      2.0       No   
3       No drinks in past 30 days     None  Excellent     None       No   
4                 2 days per week     None       Fair      2.0       No   
...                           ...      ...        ...      ...      ...   
445127        Don’t know/Not sure      3.0       Good     None       No   
445128  No drinks in past 30 days      2.0  Excellent      2.0       No   
445129        Don’t know/Not sure     30.0       Poor     30.0       No   
445130  No drinks in past 30 days     None  Very good     None       No   
445131            2 days per week      1.0  Very good     None       No   

       CVDCRHD4  
0            No  
1            No  
2            No  
3            No  
4            No  
...         ...  
445127       No  
445128       No  
445129       No  
445130       No  
445131       No  

[445132 rows x 18 columns]

----